dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.42/V2_42_28__Make_ou_event_and_enrollment_not_null.sql

```diff
@@ -0,0 +1,18 @@
+DO $$
```

Comment:
Would it make sense to first check for inconsistencies and only execute the procedure if any are found, thereby avoiding running it on all instances?

Reply:
This procedure is doing nothing else than remove null values if there is any.
It is a procedure because we need to catch the exception and raise a understandable message.

Reply:
@zubaira 

is 

```sql
    where en.organisationunitid is null
      and en.programid in (select programid from program where type = 'WITHOUT_REGISTRATION');
```

not limiting/finding the inconsistencies?

Reply:
@teleivo 
What I meant was something like this, but I don't believe it will yield much benefit.

```
BEGIN
    SELECT 
        (SELECT COUNT(*) 
         FROM enrollment 
         WHERE organisationunitid IS NULL 
           AND programid IN (SELECT programid FROM program WHERE type = 'WITHOUT_REGISTRATION'))
        +
        (SELECT COUNT(*) 
         FROM event 
         WHERE organisationunitid IS NULL)
        INTO total_nulls;

    IF total_nulls > 0 THEN
   -- the rest of the procedure goes here
```